lib/chef/knife/azurerm_base.rb

```diff
@@ -109,8 +109,21 @@ def token_details_for_linux
       def check_token_validity(token_details)
         time_difference = Time.parse(token_details[:expiry_time]) - Time.now.utc
         if time_difference <= 0
-          raise "Token has expired. Please run any XPLAT command like 'azure vm list' to get new token OR run 'azure login' command"
+          begin
+            xplat_command = Mixlib::ShellOut.new("azure vm image list-skus westus RedHat RHEL").run_command
```

Comment:
We can remove variable assignment here since we are only running the command and we don't need its output anywhere.